src/frame/js/actions/conversation.js

```diff
@@ -516,3 +514,56 @@ export function disconnectFaye() {
         dispatch(unsetFayeSubscriptions());
     };
 }
+
+function handleUserConversationResponse({appUser, conversation, sessionToken}) {
+    return (dispatch) => {
+        const actions = [
+            setUser(appUser),
+            setConversation(conversation),
+            setMessages(conversation.messages),
+            setAuth({
+                sessionToken
+            })
+        ];
+
+        dispatch(batchActions(actions));
+    };
+}
+
+export function startConversation() {
+    return (dispatch, getState) => {
+        const {user: {_id: userId, pendingAttributes}, config: {appId}, conversation: {_id: conversationId}} = getState();
+
+        if (conversationId) {
+            return Promise.resolve();
+        }
+
+        return dispatch(http('POST', userId ? `/appusers/${userId}/conversations` : `/apps/${appId}/appusers`, {
+            ...pendingAttributes,
```

Comment:
Should these be under an `appUser` field? Especially if the path is `/appusers/${userId}/conversations`, putting values like `firstName` and `properties` at the root seems weird